nim/src/related.nim

```diff
@@ -1,81 +1,66 @@
 # This is just an example to get you started. A typical binary package
 # uses this file as the main entry point of the application.
-import std/tables
-import std/sequtils
-import times
+import std/[tables, sequtils, monotimes, times]
 import jsony
 
 type
-  Post = object
+  Post = ref object
     `"_id"`: string
     title: string
     tags : seq[string]
 
-  RelatedPosts = object
+  RelatedPosts = ref object
     `"_id"`: string
     tags : seq[string]
     related : seq[Post]
 
-const 
+const
     input = "../posts.json"
     output = "../related_posts_nim.json"
 
+let posts = readFile(input).fromJson(seq[Post])
 
-let content = readFile(input)
-let posts = content.fromJson(seq[Post])
+let start = getMonotime()
 
-let start = getTime()
-
-var tagMap = initTable[string, var seq[int]]()
-
-for i, post in posts.pairs:
+var tagMap: Table[string, seq[int]]
+for i, post in posts:
   for tag in post.tags:
-    var lst = tagMap.getOrDefault(tag, @[])
-    lst.add(i)
-    tagMap[tag] = lst
+    if tagMap.hasKeyOrPut(tag, @[i]):
+      tagMap[tag].add i
```

Comment:
```suggestion
    tagMap.mgetOrPut(tag, @[]).add(i)
```

Your version is still doing a double lookup on happy case, I'd expect a single lookup to slightly improve things